main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala

```diff
@@ -444,7 +442,7 @@ object MonoDefs {
               val subst1 = caseSubst @@ subst.nonStrict
               // visit the body under the extended environment
               val body = visitExp(body0, env1, StrictSubstitution(subst1, root.eqEnv))
-              val eff = Type.mkUnion(exp.eff, body0.eff, loc.asSynthetic)
+              val eff = Type.mkUnion(e.eff, body.eff, loc.asSynthetic)
```

Comment:
I see here the old code was using the wrong (non-monomorphed) sub expressions.

Reply:
Yes, I will revisit the old type match ticket after this to see if it's magically fixed.